plugins/sharding/logical_range_filter.rb

```diff
@@ -237,23 +237,9 @@ def execute
 
           return if @filtered_result_sets.empty?
 
-          if @window
-            @filtered_result_sets.each do |result_set|
-              @window.each(result_set) do |windowed_result_set|
-                @temporary_tables << windowed_result_set
-                if @context.dynamic_columns.have_filtered?
-                  apply_targets = [[windowed_result_set]]
-                  @context.dynamic_columns.apply_filtered(apply_targets)
-                end
-                sort_result_set(windowed_result_set)
-                return if @context.current_limit.zero?
-              end
-            end
-          else
-            apply_filtered_dynamic_columns
-            @filtered_result_sets.each do |result_set|
-              sort_result_set(result_set)
-            end
+          each_result_set do |result_set|
+            sort_result_set(result_set)
+            return if @window && @context.current_limit.zero?
```

Comment:
Is `@window &&` needed?

Reply:
I had added this just because originally `@context.current_limit.zero?` was executed only when `@window`.
However, even removing `@window &&`, all `logical_range_filter` tests were passed.
So I have removed this `@window &&`.